frontend/catalyst/debug/printing.py

```diff
@@ -15,54 +15,90 @@
 """This module enables runtime printing of program values."""
 
 import builtins
+from functools import partial
 
 import jax
 
+from catalyst.api_extensions.callbacks import debug_callback
 from catalyst.jax_primitives import print_p
 from catalyst.tracing.contexts import EvaluationContext
 
 
 # pylint: disable=redefined-builtin
-def print(x, memref=False):
+def print(fmt, *args, **kwargs):
+    """A print function.
+
+    This print function works with python callbacks, so all arguments will be printed in the same
+    way as the :func:`builtins.print` function prints them.
+
+    One difference is that if the first argument is a string, it may be used similarly to an
+    fstring. Like so:
+
+    .. code-block:: python
+
+        @qjit
+        def cir(a, b, c):
+            debug.print("c={c} b={b} a={a}", a=a, b=b, c=c)
+
+        cir(1, 2, 3)
+    >>> c=3 b=2 a=1
+    """
+
+    if isinstance(fmt, str):
+        debug_callback(partial(_format_print_callback, fmt))(*args, **kwargs)
+        return
+
+    debug_callback(_print_callback)(fmt, *args, **kwargs)
```

Comment:
@erick-xanadu this is my fault for missing this, but I see that the old print function had this behaviour:

> Outside a qjit() compiled function the operation falls back to the Python print statement.

I didn't include this in the requirements as I had forgotten this. I'm curious if the new function already works like this?

Reply:
If not, no worries; I can add this as a new feature on SC

Reply:
It works!

```python
from catalyst.debug import print
print("hello world")
```